pyomo/contrib/gdpopt/GDPopt.py

```diff
@@ -76,101 +178,36 @@ def solve(self, model, **kwds):
-        solve_data.initialization_strategy = kwds.pop('init_strategy', None)
-        solve_data.custom_init_disjuncts = kwds.pop(
-            'custom_init_disjuncts', None)
-        solve_data.max_slack = kwds.pop('max_slack', 1000)
-        solve_data.OA_penalty_factor = kwds.pop('OA_penalty', 1000)
-        solve_data.nlp_solver_name = kwds.pop('nlp', 'ipopt')
-        solve_data.nlp_solver_kwargs = kwds.pop('nlp_kwargs', {})
-        solve_data.mip_solver_name = kwds.pop('mip', 'gurobi')
-        solve_data.mip_solver_kwargs = kwds.pop('mip_kwargs', {})
-        solve_data.modify_in_place = kwds.pop('solve_in_place', True)
-        solve_data.master_postsolve = kwds.pop(
-            'master_postsolve', _DoNothing())
-        solve_data.subproblem_presolve = kwds.pop(
-            'subprob_presolve', _DoNothing())
-        solve_data.subproblem_postsolve = kwds.pop(
-            'subprob_postsolve', _DoNothing())
-        solve_data.subproblem_postfeasible = kwds.pop(
-            'subprob_postfeas', _DoNothing())
-        solve_data.algorithm_stall_after = kwds.pop('algorithm_stall_after', 2)
-        solve_data.tee = kwds.pop('tee', False)
-
-        if solve_data.tee:
-            old_logger_level = logger.getEffectiveLevel()
-            logger.setLevel(logging.INFO)
-
-        if kwds:
-            logger.warn("Unrecognized arguments passed to GDPopt solver: %s"
-                        % (kwds,))
-
-        # Verify that decomposition strategy chosen is one of the supported
-        # strategies
-        valid_strategies = ['LOA']
-        if solve_data.decomposition_strategy not in valid_strategies:
-            raise ValueError('Unrecognized decomposition strategy %s. '
-                             'Valid strategies include: %s'
-                             % (solve_data.decomposition_strategy,
-                                valid_strategies))
-
-        # To accommodate multi-phase hybrid strategies, define an attribute to
-        # track the current phase.
-        if solve_data.decomposition_strategy == 'hPSC':
-            solve_data.current_strategy = 'PSC'
-        else:
-            solve_data.current_strategy = solve_data.decomposition_strategy
+        config = self.CONFIG(kwds.pop('options', {}))
 
-        # When generating cuts, small duals multiplied by expressions can cause
-        # problems. Exclude all duals smaller in absolue value than the
-        # following.
-        solve_data.small_dual_tolerance = 1E-8
-        solve_data.integer_tolerance = 1E-5
+        config.set_value(kwds)
 
-        solve_data.round_NLP_binaries = True
-        """bool: flag to round binary values to exactly 0 or 1.
-        Rounding is done before solving NLP subproblem"""
+        solve_data = GDPoptSolveData()
+
+        old_logger_level = config.logger.getEffectiveLevel()
+        if config.tee:
+            config.logger.setLevel(logging.INFO)
+        else:
+            config.logger.setLevel(logging.WARNING)
```

Comment:
Why pass a logger object if you are going to forcibly overwrite configurations?  I am OK with "tee" changing the level, but you should probably only move it IF tee is specified, and then only if the logger is not already enabled for INFO.